src/lua/zencode_array.lua

```diff
@@ -281,3 +281,26 @@ When("create the standard deviation of elements in array ''", function(arr)
 	ACK.standard_deviation = O.from_string(stats.standardDeviation(data):decimal())
 	new_codec('standard_deviation', { encoding="string" })
 end)
+
+local function _flat_array(data, res)
+   for _, item in ipairs(data) do
```

Comment:
should use `pairs()` instead of `ipairs()` here and in general best use `pairs` everywhere